### PR TITLE
Clarify compat syntax vs master syntax.

### PR DESCRIPTION
**Pull request by yuyichao**
Mention non-breaking rule in developer tips.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -229,6 +232,18 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 ## Developer tips
 
+One of the most important rules for `Compat.jl` is to avoid breaking users
+whenever possible, especially on a released version.
+
+Although the `master` syntax is the preferred compat syntax,
+there are cases where this shouldn't be used.
+Examples including when the new syntax already has a different meaning
```
Comment (tkelman) [inline README.md]: include

File: README.md [hunk 2]
```diff
@@ -237,7 +237,7 @@ whenever possible, especially on a released version.
 
 Although the `master` syntax is the preferred compat syntax,
 there are cases where this shouldn't be used.
-Examples includ when the new syntax already has a different meaning
```
Comment (yuyichao) [inline README.md]: Ooops

File: README.md [hunk 3]
```diff
@@ -229,6 +232,18 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 ## Developer tips
 
+One of the most important rules for `Compat.jl` is to avoid breaking users
```
Comment (stevengj) [inline README.md]: breaking user code?

File: README.md [hunk 4]
```diff
@@ -229,6 +232,18 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 ## Developer tips
 
+One of the most important rules for `Compat.jl` is to avoid breaking user code
+whenever possible, especially on a released version.
+
+Although the `master` syntax is the preferred compat syntax,
+there are cases where this shouldn't be used.
+Examples include when the new syntax already has a different meaning
+on previous versions of julia, or when functions are removed from `Base`
```
Comment (nalimilan) [inline README.md]: Julia (same below)

File: README.md [hunk 5]
```diff
@@ -229,6 +232,18 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 ## Developer tips
 
+One of the most important rules for `Compat.jl` is to avoid breaking user code
+whenever possible, especially on a released version.
+
+Although the `master` syntax is the preferred compat syntax,
```
Comment (nalimilan) [inline README.md]: Maybe `git master` or "syntax used in the most recent Julia version"?

Comment (yuyichao) [inline README.md]: This was copied from above, but sure..

File: README.md (hunk 4 above)
Comment (yuyichao) [inline README.md]: Done. Also replaced a few other `julia`s above for consistency

Comment (nalimilan) [inline README.md]: Wow, I hadn't noticed there were so many of them. :-)

Comment (timholy) [inline README.md]: There has sometimes been a distinction made between Julia the language and julia the implementation of the language. In the case of Compat, it's probably the former you mean, so this seems reasonable.

**Pull request merged**